src/components/NACWidget/index.tsx

```diff
@@ -16,30 +61,16 @@ export function NACWidget({
   widgetsVersion: string
   widgetsBaseUrl: string
 }) {
-  const containerRef = useRef<HTMLDivElement>(null)
-  const pathname = usePathname()
-  const [instanceId] = useState(() => Math.random().toString(36).substring(2, 9))
-  const scriptRefsRef = useRef<HTMLScriptElement[]>([])
-
-  useEffect(() => {
-    if (!containerRef.current) return
-
-    containerRef.current.innerHTML = ''
+  const widgetId = `nac-widget-${widget}`
 
-    const widgetId = `nac-widget-${widget}-${instanceId}`
+  // Removes any trailing slashes
+  const safeBaseUrl = widgetsBaseUrl.replace(/\/+$/, '')
 
-    const appDiv = document.createElement('div')
-    appDiv.id = widgetId
-    containerRef.current.appendChild(appDiv)
+  const scriptUrl = `${safeBaseUrl}/${widgetsVersion}`
 
-    // Removes any trailing slashes
-    const safeBaseUrl = widgetsBaseUrl.replace(/\/+$/, '')
-
-    const scriptUrl = `${safeBaseUrl}/${widgetsVersion}`
-
-    // Base URL (used for Google Analytics)
-    const baseUrl = window.location.pathname
+  const { scriptName, widgetDataKey, widgetControllerKey } = loadersByWidget[widget]
 
+  useEffect(function loadWidgetCss() {
     const loadCSS = (url: string): Promise<string> => {
```

Comment:
Is there any downside/reason to declare this function definition inside of the useEffect as opposed to just at the top-level scope of this file? We're not closing over any state that I can tell.